### PR TITLE
Changed MenuItem to ToggleButton (#2776)

### LINKED ISSUE #2776: Replace display mode select to toggle button
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Like https://github.com/mui/mui-toolpad/pull/2764, replace the existing drop-down component with a more suitable toggle button component.

Screenshot 2023-10-10 at 8 24 31 AM

### Examples 🌈

_No response_

### Motivation 🔦

_No response_

### PR DESCRIPTION
**Pull request by biplobsd**
I have changed MenuItem to ToggleButton as in the #2764 discussed.

- [x] I've read and followed the [contributing guide](https://github.com/mui/mui-toolpad/blob/master/CONTRIBUTING.md#sending-a-pull-request) on how to create great pull requests.
- [ ] I've updated the relevant documentation for any new or updated feature.
- [x] I've linked relevant GitHub issue with "Closes #2776".
- [x] I've added a visual demonstration in the form of a screenshot or video.

## Screenshort
image

### CODE REVIEW

**Comment by biplobsd** [comment]
@apedroferreira Kindly take a review. Thanks

**Review by apedroferreira — CHANGES_REQUESTED** [review]
Awesome, thank you so much, it's perfect!

Before we can merge, can you just please:
1. Update the image in https://mui.com/toolpad/concepts/display-mode/
1. Run `yarn prettier:all` to fix the prettier errors
1. Fix the type error reported below in `test_types` too? We need to adjust the type for the `onChange` prop.

You can ignore the `test_unit` below, we will fix it. Thanks a lot!

**Comment by biplobsd** [comment]
@apedroferreira Could you explain more the step 3. Is it in the CirecleCI `test_types` report?

**Comment by apedroferreira** [comment]
> @apedroferreira Could you explain more the step 3. Is it in the CirecleCI `test_types` report?

Yes. that was it.
All good, now, thanks again!

**Review by apedroferreira — APPROVED** [review]

File: packages/toolpad-app/src/toolpad/AppEditor/PageEditor/PageOptionsPanel.tsx [hunk 1]
```diff
@@ -21,14 +29,14 @@ export default function PageOptionsPanel() {
   const page = appDom.getNode(dom, pageNodeId, 'page');
 
   const handleDisplayModeChange = React.useCallback(
-    (event: React.ChangeEvent<HTMLInputElement>) => {
+    (event: React.MouseEvent<HTMLElement>, newValue: string) => {
```
Comment (apedroferreira) [inline packages/toolpad-app/src/toolpad/AppEditor/PageEditor/PageOptionsPanel.tsx:32]:
```suggestion
    (event: React.MouseEvent<HTMLElement>, newValue: appDom.PageDisplayMode) => {
```

Just noticed I think we don't need to use `as`, we can just do this.

File: packages/toolpad-app/src/toolpad/AppEditor/PageEditor/PageOptionsPanel.tsx [hunk 2]
```diff
@@ -21,14 +29,14 @@ export default function PageOptionsPanel() {
...
       domApi.update((draft) =>
         appDom.setNodeNamespacedProp(
           draft,
           page,
           'attributes',
           'display',
-          event.target.value as appDom.PageDisplayMode,
+          newValue as appDom.PageDisplayMode,
```
Comment (apedroferreira) [inline packages/toolpad-app/src/toolpad/AppEditor/PageEditor/PageOptionsPanel.tsx:39]:
```suggestion
          newValue,
```

**Comment by biplobsd** [comment]
@apedroferreira Can I get your Twitter handle? I want to tweet about this experience. 🙂

**Comment by apedroferreira** [comment]
> @apedroferreira Can I get your Twitter handle? I want to tweet about this experience. 🙂

I don't have a Twitter, but feel free to do it, maybe you can tag MUI or something :)

**Pull request merged**